### PR TITLE
[http3] Add intra-cluster forwarding event counter 

### PR DESCRIPTION
**Pull request by hfujita**
This PR adds two event counters, `http3.packet-forwarded` and
`http3.forwarded-packet-received` to track intra-cluster QUIC packet
forwarding.

### CODE REVIEW

**Review by deweerdt — APPROVED** [review]
Thank you!

**Review by kazuho — APPROVED** [review]
Thank you for the PR. LGTM.

File: t/40http3-forward.t [hunk 1]
```diff
@@ -46,6 +47,21 @@ is do {local $/; join "", <$slow_fh>}, "server=1", "slow request succeeded";
...
+my $resp = `curl --silent -o /dev/stderr http://127.0.0.1:${server1_port}/server-status/json?show=events 2>&1 > /dev/null`;
+my $jresp = decode_json("$resp");
+my $num_forwarded_received = $jresp->{'http3.forwarded-packet-received'};
+
+$resp = `curl --silent -o /dev/stderr http://127.0.0.1:${server2_port}/server-status/json?show=events 2>&1 > /dev/null`;
+$jresp = decode_json("$resp");
+my $num_forwarded = $jresp->{'http3.packet-forwarded'};
+
+ok($num_forwarded > 0, "some packets were forwarded through event counter");
+ok($num_forwarded == $num_forwarded_received, "packets forwarded == packets received");
```
Comment (kazuho) [inline t/40http3-forward.t:63]:
```suggestion
cmp_ok($num_forwarded, '>', 0, "some packets were forwarded through event counter");
is($num_forwarded, num_forwarded_received, "packets forwarded == packets received");
```
Using these test functions would give us more debugging information when things go wrong.

**Review by kazuho — APPROVED** [review]
Thank you for the fix.

As discussed elsewhere, the existing design flaw is that we are forwarding QUIC datagrams where we should be forwarding UDP datagrams.

There are certain requirements in QUIC that require UDP datagrams to have certain properties (e.g., datagrams carrying ack-eliciting Initial packets have to be full-sized).  I do not think current design of forwarding would interfere from QUIC v1 working correctly, but we'd better streamline our implementation (to do forwarding at the datagram-level).

**Pull request merged**